controllers/admin/AdminCartRulesController.php

```diff
@@ -200,10 +200,44 @@ public function postProcess()
                 }
             }
 
-            // These are checkboxes (which aren't sent through POST when they are not check), so they are forced to 0
+            // These are checkboxes (which aren't sent through POST when they are not checked), so they are forced to 0
+            $restrictions_selected = false;
             foreach (['country', 'carrier', 'group', 'cart_rule', 'product', 'shop'] as $type) {
                 if (!Tools::getValue($type . '_restriction')) {
                     $_POST[$type . '_restriction'] = 0;
+                } elseif ($type !== 'shop') { // shop restriction is forced to be checked, thus we do not take it into account
+                    $restrictions_selected = true;
+                }
+            }
+
+            // At least one of these is compulsory: code, customer or any restriction. If none of these are present, the cart rule applies to all carts !
+            if (empty(Tools::getValue('id_cart_rule')) &&
+                empty($restrictions_selected) &&
```

Comment:
Is there a reason to use `empty($restrictions_selected)` instead of `!$restrictions_selected` ?

Reply:
For me it is more consistent to use `!$restrictions_selected` because it respects the type. $restrictions_selected a local context boolean, it should not become anything else but using empty make me think it could be something else because I often use it to manage polymorphism or undefined objects.

Whatever it is not blocking for me. Feel free to solve my comment :)